src/lib/ServiceClient.js

```diff
@@ -137,9 +137,9 @@ class ServiceClient extends EventEmitter {
         oldCall.reject(err);
       }
 
-      // if there weren't any other calls in the queue, execute this new call
+      // if there weren't any other calls in the queue and there's no current call, execute this new call
       // otherwise new call will be handled in order when others complete
-      if (this._callQueue.length === 1) {
+      if (this._callQueue.length === 1 && this._currentCall === null) {
```

Comment:
I think we're ok here - `_executeCall` checks if we're shutdown and any call ending (successful or otherwise) should trigger the client to see if anything is queued and execute that call.